sound/soc/sof/intel/probe.c

```diff
@@ -0,0 +1,199 @@
+// SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause)
+//
+// This file is provided under a dual BSD/GPLv2 license.  When using or
+// redistributing this file, you may do so under either license.
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+//
+// Author: Cezary Rojewski <cezary.rojewski@intel.com>
+//
+
+#include <sound/hdaudio_ext.h>
+#include "../sof-priv.h"
+#include "probe.h"
+
+int sof_ipc_probe_init(struct snd_sof_dev *sdev,
+		size_t buffer_size)
+{
+	struct sof_ipc_probe_dma_set_params *msg;
+	struct sof_ipc_reply reply;
+	unsigned int stream_tag = INVALID_NODE_ID;
```

Comment:
except for this maybe, this looks a bit too Intel specific.

Reply:
After even more thinking, only _init method would have to be updated to be no longer HDA dependent. Right now I'm using sdev->extractor directly.